apps/prairielearn/assets/scripts/instructorAssessmentAccessClient.ts

```diff
@@ -0,0 +1,216 @@
+        timeLimitMin: parseInt(accessRule.assessment_access_rule.time_limit_mins),
+        password: accessRule.assessment_access_rule.password,
+        examUuid: accessRule.assessment_access_rule.exam_uuid,
+      };
+      const filteredRules = Object.fromEntries(Object.entries(rule).filter(([_, value]) => value));
+      return filteredRules;
+    });
+    assessmentAccessRulesInput.value = JSON.stringify(accessRulesMap);
+    form.submit();
+  });
+
+  function adjustedDate(dateString: string | Date) {
+    const date = new Date(dateString);
+    const timezoneOffset = date.getTimezoneOffset() * 60000;
+    return new Date(date.getTime() - timezoneOffset);
+  }
+
+  function handleUpdateAccessRule({ form }: { form: HTMLFormElement }) {
+    const formData = new FormData(form);
+    const updatedAccessRules: Record<string, any> = Object.fromEntries(formData);
+    const row = parseInt(updatedAccessRules.row.toString());
+    updatedAccessRules.uids = updatedAccessRules.uids
+      .toString()
+      .split(',')
+      .map((uid: string) => uid.trim());
+    updatedAccessRules.active = updatedAccessRules.active === 'true';
+
+    updatedAccessRules.start_date = new Date(
+      Temporal.PlainDateTime.from(updatedAccessRules.start_date)
+        .toZonedDateTime(timezone)
+        .toInstant().epochMilliseconds,
+    ).toISOString();
+
+    updatedAccessRules.end_date = new Date(
+      Temporal.PlainDateTime.from(updatedAccessRules.end_date)
+        .toZonedDateTime(timezone)
+        .toInstant().epochMilliseconds,
+    ).toISOString();
+
+    if (accessRulesData[row]) {
+      accessRulesData[row].assessment_access_rule = updatedAccessRules;
+    } else {
+      accessRulesData[row] = { assessment_access_rule: updatedAccessRules };
+    }
+    refreshTable();
+  }
+
+  on('click', '.editButton', (e) => {
+    const editButton = (e.target as HTMLElement).closest('button');
+    if (!editButton) return;
+
+    templateFromAttributes(editButton, editAccessRuleModal as HTMLElement, {
+      'data-row': '.access-rule-row',
+      'data-title-text': '.modal-title',
+      'data-submit-text': '.updateAccessRuleButton',
+      'data-access-rule-mode': '.access-rule-mode',
+      'data-access-rule-uids': '.access-rule-uids',
+      'data-access-rule-start-date': '.access-rule-start-date',
+      'data-access-rule-end-date': '.access-rule-end-date',
+      'data-access-rule-active': '.access-rule-active',
+      'data-access-rule-credit': '.access-rule-credit',
+      'data-access-rule-time-limit': '.access-rule-time-limit',
+      'data-access-rule-password': '.access-rule-password',
+      'data-access-rule-exam-uuid': '.access-rule-exam-uuid',
+    });
+
+    $('#editAccessRuleModal').modal('show');
+  });
+
+  on('click', '#updateAccessRuleButton', (e) => {
+    const form = (e.target as HTMLElement).closest('form');
+    if (!form) return;
+    handleUpdateAccessRule({ form });
+  });
+
+  function swapRows(row: any, targetRow: any) {
```

Comment:
Is this correct?

```suggestion
  function swapRows(row: number, targetRow: number) {
```

Reply:
Yes, updated to this locally in [660f81b](https://github.com/PrairieLearn/PrairieLearn/pull/9934/commits/660f81bebe37d51fcef94b1535961f4abfa02d88).